File: array/27/27/27.cpp
#include <iostream>
#include <vector>

using namespace std;

int removeElement(vector<int>& nums, int val) {
	if (nums.size() == 0) {
		return 0;
	}
	if (nums.size() == 1) {
		if (nums[0] == val) {
			return 0;
		}
		else {
			return 1;
		}
	}

	int slow = 0;
	for (int fast = 0; fast < nums.size(); fast++) {
		if (nums[fast] != val) {
			nums[slow] = nums[fast];
			slow++;
		}
	}
	return slow;

}
int main(){
	vector<int> nums{ 0, 1, 2, 2, 3, 0, 4, 2 };
	int val = 2;
	int len = removeElement(nums, val);
	cout << len<<endl;
	for (int i = 0; i < len; i++) {
		cout<<nums[i];
	}
}

File: array/27.py
class Solution:
    def removeElement(self, nums, val: int) -> int:
        i, j = 0, 0
        for j in range(0, len(nums)):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i






if __name__ == '__main__':
    s = Solution()
    nums = [3]
    val = 2
    i = s.removeElement(nums, val)
    print(i)
    print(nums[:i])